### ISSUE TITLE
Crash on bad input data

### ISSUE THREAD

**Issue by kumpera** [VowpalWabbit/vowpal_wabbit#2719]
#### Describe the bug
VW crashes on bad input data.

#### To Reproduce
From `build/vowpal_wabbit`: `./vw --cb_explore_adf --cb_type ips --dsjson  ../../test/train-sets/0001.json`

**Comment by lalo** [comment]
By bad data you mean that the correct usage of that file would be --json instead of --dsjson, right? Anyways, it shouldn't crash.

**Comment by jackgerrits** [comment]
```
./vw -- --cb_explore_adf --cb_type ips --dsjson ../../test/train-sets/0001.json
~~~~~^
```
What is the significance of the `--`, is VW trying to open this as the datafile?

**Comment by kumpera** [comment]
@jackgerrits by bad, I C&P the command line from lldb. It should not be there. Fixed the description.
@lalo vw should not crash, ever. Bad input should produce errors.

**Comment by jackgerrits** [comment]
What is the crash? Segfault or exception?

**Comment by zwd-ms** [comment]
Hey Jack,

Do we have a way to reproduce crash?

Thanks,
Wangda

**Comment by zwd-ms** [comment]
Tested with vw 8.10.0 on WSL. I don't see crash, but apparently there's no appropriate warning message.

~~~sh
./vw --cb_explore_adf --cb_type ips --dsjson ../../test/train-sets/0001.json
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = ../../test/train-sets/0001.json
WARNING: Old string feature value behavior is deprecated in JSON/DSJSON and will be removed in a future version. Use `--chain_hash` to use new behavior and silence this warning.
num sources = 1
Enabled reductions: gd, scorer, csoaa_ldf, cb_adf, cb_explore_adf_greedy, shared_feature_merger
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features

finished run
number of examples = 0
weighted example sum = 0.000000
weighted label sum = 0.000000
average loss = n.a.
total feature number = 0
~~~

The command-line doesn't look right. Changing `--dsjson` to `--json` and removing `--cb_explore_adf` to match the data file:

~~~sh
./vw --cb_type ips --json ../../test/train-sets/0001.json
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = ../../test/train-sets/0001.json
WARNING: Old string feature value behavior is deprecated in JSON/DSJSON and will be removed in a future version. Use `--chain_hash` to use new behavior and silence this warning.
num sources = 1
Enabled reductions: gd, scorer
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
1.000000 1.000000            1            1.0   1.0000   0.0000       51
0.513618 0.027236            2            2.0   0.0000   0.1650      104
0.263121 0.012624            4            4.0   0.0000   0.0569      135
0.237739 0.212356            8            8.0   0.0000   0.2024      146
0.242021 0.246303           16           16.0   1.0000   0.3249       24
0.235878 0.229736           32           32.0   0.0000   0.2256       32
0.230921 0.225964           64           64.0   0.0000   0.1601       61
0.223511 0.216101          128          128.0   1.0000   0.8308      106

finished run
number of examples = 200
weighted example sum = 200.000000
weighted label sum = 91.000000
average loss = 0.195760
total feature number = 15482
~~~

@jackgerrits could you confirm what the desired behavior/message should be in this case?

**Comment by jackgerrits** [comment]
In this case the desired behavior is that we don't crash as was reported in the original bug. If we simply do not process any events, then I think this is okay? I am not sure if we have enough information to provide a relevant warning here

**Comment by zwd-ms** [comment]
ok then, confirmed no crash.
